### PR TITLE
Initial skeleton structure

### PR DESCRIPTION
**Pull request by soderluk**
Add initial structure for the Translator skeleton
Update readme
Add controller actions
Support for validating signatures and signing data

### CODE REVIEW

File: app/controllers.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    _service = None  # type: services.Translator
+
+    def __init__(self):
+        """Initializes the translator controller"""
+        self._service = services.Translator()
+
+    @request_args({
+        'timestamp': fields.Str(required=True),
+        'productCode': fields.Str(required=True),
+        'parameters': fields.Dict(allow_none=True),
```
Comment (lietu) [inline app/controllers.py]: Is it possible to define more specific validation here - like add some dummy field in the `parameters` object like "name" and validate it's a string?

File: app/controllers.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        :return: The translator data.
+        :rtype: responses.JSONResponse
+        :raise ValidationFailed: If the validation fails.
+        """
+        # Validate security
+        try:
+            utils.validate(bottle.request.headers, args)
+        except bottle.HTTPError:
+            logger.exception('Validation failed.')
+            raise ValidationFailed('Validation failed.')
```
Comment (lietu) [inline app/controllers.py:89]: How about just `raise`?

**Review by edanchenkov — APPROVED** [review]

**Review by lietu — CHANGES_REQUESTED** [review]
A few small changes please

File: app/controllers.py (hunk 2 above)
Comment (soderluk) [reply app/controllers.py:89]: The ValidationFailed() is the correct one to raise here.

File: app/controllers.py (hunk 1 above)
Comment (soderluk) [reply app/controllers.py]: Added name as mandatory extra parameter

**Review by lietu — APPROVED** [review]

**Pull request merged**